### PR TITLE
Support sort_nulls_last mode

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

For TPC-DS test easy
1. Add config to set default nulls order

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/OrderByElement.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Comment by Seaven** [comment]
run starrocks_be_unittest

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -337,7 +337,7 @@
     @VarAttr(name = CBO_CTE_REUSE)
     private boolean cboCteReuse = false;
 
-    @VarAttr(name = CBO_CTE_REUSE_RATE, flag = VariableMgr.INVISIBLE)
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:340]: This also can be changed even if it is invisible

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/OrderByElement.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

**Comment by HangyuanLiu** [comment]
I don't think we should add config randomly for testing convenience, it will bring unpredictable confusion

**Comment by Seaven** [comment]
> I don't think we should add config randomly for testing convenience, it will bring unpredictable confusion

I aggree with you, but modifying a large number of SQL is too slow, and we may also need to compare with other systems in the future

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/OrderByElement.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|2|2|100.00%|

**Review by Youngwb — DISMISSED** [review]

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/SqlParser.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/OrderByElement.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/analysis/SortInfo.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/SqlModeHelper.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/QueryTransformer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/WindowTransformer.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by Seaven** [comment]
@Mergifyio backport branch-2.2

**Pull request merged**